src/abstractdataframe/abstractdataframe.jl

```diff
@@ -530,19 +530,24 @@ julia> describe(df, :min, :sum => sum)
 │ 2   │ x        │ 0.1 │ 5.5 │
 │ 3   │ y        │ 'a' │     │
 
-julia> describe(df, :min, :sum => sum, cols=:x)
+julia> describe(df, cols=:x, :min, sum => :sum)
 1×3 DataFrame
 │ Row │ variable │ min     │ sum     │
 │     │ Symbol   │ Float64 │ Float64 │
 ├─────┼──────────┼─────────┼─────────┤
 │ 1   │ x        │ 0.1     │ 5.5     │
 ```
 """
-DataAPI.describe(df::AbstractDataFrame,
-                 stats::Union{Symbol, Pair{<:SymbolOrString}}...;
-                 cols=:) =
-    _describe(select(df, cols, copycols=false), collect(stats))
-
+function DataAPI.describe(df::AbstractDataFrame, stats::Union{Symbol,
+                          Pair{<:Base.Callable,<:SymbolOrString},
+                          Pair{<:SymbolOrString}}...; # TODO: remove after deprecation
+                 cols=:)
+    if any(x -> x isa Pair{<:SymbolOrString}, stats)
+        Base.depwarn("name => function order is deprecated; use function => name instead", :describe)
+    end
+    return _describe(select(df, cols, copycols=false),
+                     [s isa Pair{<:SymbolOrString} ? last(s) => first(s) : s for s in stats])
```

Comment:
Maybe better for latency?
```suggestion
                     Any[s isa Pair{<:SymbolOrString} ? last(s) => first(s) : s for s in stats])
```

Reply:
and safer in general :smile: